packages/api-client-core/src/types.ts

```diff
@@ -764,3 +775,92 @@ export type PaginateOptions = {
 export type AvailableSelection<Schema> = Schema extends string | number | bigint | null | undefined
   ? boolean | null | undefined
   : { [key in keyof Schema]?: AvailableSelection<Schema[key]> };
+
+/** Options for configuring a concurrency group for a background action */
+export interface BackgroundActionConcurrencyGroup {
+  /**
+   * The name of the group to put this background action in. If more than the `maxConcurrency` number of actions are enqueued into a group, they will wait to execute until one of the earlier ones has finished.
+   */
+  key: string;
+
+  /**
+   * The maximum number of actions in this group that can be running at once
+   */
+  maxConcurrency?: number;
+}
+
+/**
+ * Options for governing how a background action is enqueued
+ */
+export type EnqueueBackgroundActionOptions<Action extends AnyActionFunction> = {
+  /**
+   * A unique identifier to use for this background action. Must be unique among all other background actions within this environment. If not set, a unique ID will be autogenerated and returned.
+   **/
+  id?: string;
+
+  /**
+   * Which backend queue to enqueue this action into. If not set, the `default` queue will be used.
+   */
+  queue?: string;
+
+  /**
+   * A group to put this background action in that limits the maximum concurrency of all actions in the group. If not set, the action won't be concurrency limited.
+   */
+  concurrencyGroup?: string | BackgroundActionConcurrencyGroup;
+
+  /**
+   * How many times, and how fast to retry this action if it fails
+   * Matches the options you might pass to the `retry` module from npm
+   *
+   * Setting `retries` to one means the action will be attempted once, then retried once if it fails. Setting `retries` to 0 means failures won't be retried at all.
+   *
+   * @example
+   * // retry up to 3 more times after failure (4 attempts total)
+   * retry: 3
+   *
+   * @example
+   * retry: { retries: 5, minTimeout: 1000, maxTimeout: 60000, randomize: true }
+   *
+   * @default 10 retry 10 times with exponential backoff
```

Comment:
I go back and forth whether, as a user, I'd want 10 retries or 2 by default.

If my background action was "send an email to a customer", and it succeeded because it passed on the 9th retry I'd be pretty happy. However, if I wrote a bug that accessed a null pointer after successfully sending the email I'd be pretty pissed... 🤷